tst/EnergyPlus/unit/LowTempRadiantSystem.unit.cc

```diff
@@ -4663,3 +4663,135 @@ TEST_F(LowTempRadiantSystemTest, VariableFlowCoolingOnlyInputTest)
+    qRadSysSource = 0.0;
+    UpdateRadSysSourceValAvg(*state, isItOn);
+    EXPECT_TRUE(isItOn);
+    EXPECT_NEAR(qRadSysSource(1), 100.0, errTol);
+    EXPECT_NEAR(qRadSysSource(2), 200.0, errTol);
+    EXPECT_NEAR(qRadSysSource(3), 300.0, errTol);
+    EXPECT_NEAR(qRadSysSource(4), 300.0, errTol);
+    EXPECT_NEAR(qRadSysSource(5), 0.0, errTol);
+    EXPECT_NEAR(qRadSysSource(6), 0.0, errTol);
+
+    // Test 3: Only Constant Flow Radiant System--should come back with flag true and only constant flow variables set in QRadSysSource
+    isItOn = false;
+    dataLTRS->NumOfHydrLowTempRadSys = 0;
+    dataLTRS->NumOfCFloLowTempRadSys = 1;
+    dataLTRS->NumOfElecLowTempRadSys = 0;
+    dataLTRS->TotalNumOfRadSystems = dataLTRS->NumOfHydrLowTempRadSys + dataLTRS->NumOfCFloLowTempRadSys + dataLTRS->NumOfElecLowTempRadSys;
+    qRadSysSource = 0.0;
+    UpdateRadSysSourceValAvg(*state, isItOn);
+    EXPECT_TRUE(isItOn);
+    EXPECT_NEAR(qRadSysSource(1), 0.0, errTol);
+    EXPECT_NEAR(qRadSysSource(2), 0.0, errTol);
+    EXPECT_NEAR(qRadSysSource(3), 0.0, errTol);
+    EXPECT_NEAR(qRadSysSource(4), 0.0, errTol);
+    EXPECT_NEAR(qRadSysSource(5), 400.0, errTol);
+    EXPECT_NEAR(qRadSysSource(6), 0.0, errTol);
+
+    // Test 4: Only Electric Radiant System--should come back with flag true and only electric variables set in qRadSysSrc (QRadSysSource)
+    isItOn = false;
+    dataLTRS->NumOfHydrLowTempRadSys = 0;
+    dataLTRS->NumOfCFloLowTempRadSys = 0;
+    dataLTRS->NumOfElecLowTempRadSys = 1;
+    dataLTRS->TotalNumOfRadSystems = dataLTRS->NumOfHydrLowTempRadSys + dataLTRS->NumOfCFloLowTempRadSys + dataLTRS->NumOfElecLowTempRadSys;
+    qRadSysSource = 0.0;
+    UpdateRadSysSourceValAvg(*state, isItOn);
+    EXPECT_TRUE(isItOn);
+    EXPECT_NEAR(qRadSysSource(1), 0.0, errTol);
+    EXPECT_NEAR(qRadSysSource(2), 0.0, errTol);
+    EXPECT_NEAR(qRadSysSource(3), 0.0, errTol);
+    EXPECT_NEAR(qRadSysSource(4), 0.0, errTol);
+    EXPECT_NEAR(qRadSysSource(5), 0.0, errTol);
+    EXPECT_NEAR(qRadSysSource(6), 500.0, errTol);
+
+    // Test 5: All Radiant System--should come back with flag true and all variables set in qRadSysSrc (QRadSysSource)
+    isItOn = false;
+    dataLTRS->NumOfHydrLowTempRadSys = 2;
+    dataLTRS->NumOfCFloLowTempRadSys = 1;
+    dataLTRS->NumOfElecLowTempRadSys = 1;
+    dataLTRS->TotalNumOfRadSystems = dataLTRS->NumOfHydrLowTempRadSys + dataLTRS->NumOfCFloLowTempRadSys + dataLTRS->NumOfElecLowTempRadSys;
+    qRadSysSource = 0.0;
+    UpdateRadSysSourceValAvg(*state, isItOn);
+    EXPECT_TRUE(isItOn);
+    EXPECT_NEAR(qRadSysSource(1), 100.0, errTol);
+    EXPECT_NEAR(qRadSysSource(2), 200.0, errTol);
+    EXPECT_NEAR(qRadSysSource(3), 300.0, errTol);
+    EXPECT_NEAR(qRadSysSource(4), 300.0, errTol);
+    EXPECT_NEAR(qRadSysSource(5), 400.0, errTol);
+    EXPECT_NEAR(qRadSysSource(6), 500.0, errTol);
+}
```

Comment:
Seems well tested. Is a little hard on the eyes to follow, but I do like the unique numerics to track what's happening.